icu4c/source/common/rbbi.cpp

```diff
@@ -912,20 +912,18 @@ int32_t RuleBasedBreakIterator::handleNext() {
             (tableData + tableRowLen * state);
 
 
-        if (row->fAccepting == -1) {
+        uint16_t accepting = row->fAccepting;
+        if (accepting == 1) {
             // Match found, common case.
             if (mode != RBBI_START) {
                 result = (int32_t)UTEXT_GETNATIVEINDEX(&fText);
             }
-            fRuleStatusIndex = row->fTagIdx;   // Remember the break status (tag) values.
-        }
-
-        int16_t completedRule = row->fAccepting;
-        if (completedRule > 0) {
+            fRuleStatusIndex = row->fTagsIdx;   // Remember the break status (tag) values.
+        } else if (accepting >= 2) {
```

Comment:
... then this could be
  `} else if (accepting > ACCEPTING_UNCONDITIONAL) {`

Alternatively/optionally, you could check for == 0 first, then ACCEPTING_UNCONDITIONAL, then just "else"

Reply:
Thanks. (And when I write "optional" I mean it :-)